src/vscode-dts/vscode.proposed.chatParticipantAdditions.d.ts

```diff
@@ -81,7 +81,19 @@ declare module 'vscode' {
 
 	export class ChatPrepareToolInvocationPart {
 		toolName: string;
-		constructor(toolName: string);
+		/**
+		 * Partial arguments that have streamed in for the tool invocation.
+		 */
+		streamData?: ChatToolInvocationStreamData;
+		constructor(toolName: string, streamData?: ChatToolInvocationStreamData);
+	}
+
+	export interface ChatToolInvocationStreamData {
```

Comment:
Maybe you need a tool call ID here, so we know when the real tool invocation has started? Not sure how that works now

Reply:
Yeah it just gets hidden when more content shows up. I figured for parallel tool calls we can just render the last one and don't necessarily need parallel progress. This is less about showing the progress to the user and more about showing something that doesn't feel like it's stuck in a working state